src/sql/sql.ml

```diff
@@ -49,7 +49,16 @@ let sql_pool ?size uri =
         'sqlite' is not a valid scheme; did you mean 'sqlite3'?");
     let pool =
       let pool_config = Caqti_pool_config.create ?max_size:size () in
-      Caqti_lwt_unix.connect_pool ~pool_config ~post_connect parsed_uri in
+      Caqti_lwt_unix.connect_pool ~pool_config ~post_connect:(fun db ->
+        let%lwt result = standard_post_connect db in
```

Comment:
This can be just

```ocaml
match%lwt standard_post_connect db with
| Ok () ->
  (match post_connect with
  | Some f -> f db
  | None -> Lwt_result.return ())
| Error e -> Lwt_result.fail e)
```

Reply:
Or maybe like this
```ocaml
Lwt_result.bind (standard_post_connect db) (fun () ->
    match post_connect with
    | Some f -> f db
    | None -> Lwt_result.return ()))
```